parser.y

```diff
@@ -9791,7 +9791,7 @@ NumericType:
 		// TODO: check flen 0
 		x := types.NewFieldType($1.(byte))
 		x.Flen = $2.(int)
-		if $2.(int) != types.UnspecifiedLength {
+		if $2.(int) != types.UnspecifiedLength && types.TiDBStrictIntegerDisplayWidth {
```

Comment:
Why change here?

Reply:
this my previous pr, it will parse int(10) as warnings, it should be under the control of this variable